### DISCUSSION TITLE
Cors blocking using Cloud Flare  workers

### DISCUSSION THREAD

**Discussion by rogeriopavan** [honojs/hono#3315] [General]
I add the code above to allow localhost access to my worker, and its worker well.

```typescript
app.use("*", async (c, next) => {
  const corsMiddleware = cors({
    origin: ["http://localhost:3000"],
    allowMethods: ["GET", "OPTIONS", "POST", "PUT", "DELETE"],
    credentials: true,
  });
  return corsMiddleware(c, next);
});

```
Now i trying use my next.js website deployed in the cloudFlare pages and i add cors policy:

```typescript
app.use("*", async (c, next) => {
  const corsMiddleware = cors({
    origin: [
      "http://localhost:3000",
      "https://develop.scheduling-next-front-end.pages.dev",
    ],
    allowMethods: ["GET", "OPTIONS", "POST", "PUT", "DELETE"],
    credentials: true,
  });
  return corsMiddleware(c, next);
});
```

But its not working. And i received BLOCK BY CORS POLICY.
![image](https://github.com/user-attachments/assets/02cba01f-d8a8-497a-badd-9d8736f23471)

**Comment by therealjordanlee** [comment]
Hello @pavandasilva,

A typical request should include the origin header, e.g.:

![image](https://github.com/user-attachments/assets/79783c74-bb7b-467b-a9b5-e7cb96b6efd7)

This is the header which the cors middleware checks. The request in your screenshot doesn't have this header.
